eth/beacon/db/chain.py

```diff
@@ -0,0 +1,450 @@
+        and changed "header" to "block".
+        """
+        try:
+            encoded_score = db[SchemaV1.make_block_hash_to_score_lookup_key(block_hash)]
+        except KeyError:
+            raise BlockNotFound("No block with hash {0} found".format(
+                encode_hex(block_hash)))
+        return rlp.decode(encoded_score, sedes=rlp.sedes.big_endian_int)
+
+    def block_exists(self, block_hash: Hash32) -> bool:
+        """
+        NOTE: copied from ``eth.db.header.HeaderDB.header_exists``,
+        and changed "header" to "block".
+        """
+        return self._block_exists(self.db, block_hash)
+
+    @staticmethod
+    def _block_exists(db: BaseDB, block_hash: Hash32) -> bool:
+        """
+        NOTE: copied from ``eth.db.header.HeaderDB._header_exists``,
+        and changed "header" to "block".
+        """
+        validate_word(block_hash, title="Block Hash")
+        return block_hash in db
+
+    def persist_block_chain(
+        self,
+        headers: Iterable[BaseBeaconBlock]
+    ) -> Tuple[Tuple[BaseBeaconBlock, ...], Tuple[BaseBeaconBlock, ...]]:
+        """
+        Return two iterable of blocks, the first containing the new canonical blocks,
+        the second containing the old canonical headers
+
+        NOTE: copied from ``eth.db.header.HeaderDB.persist_header_chain``,
+        and changed "header" to "block".
+        """
+        with self.db.atomic_batch() as db:
+            return self._persist_block_chain(db, headers)
+
+    @classmethod
+    def _persist_block_chain(
+            cls,
+            db: BaseDB,
+            blocks: Iterable[BaseBeaconBlock]
+    ) -> Tuple[Tuple[BaseBeaconBlock, ...], Tuple[BaseBeaconBlock, ...]]:
+        """
+
+        NOTE: copied from ``eth.db.header.HeaderDB._persist_header_chain``,
+        and changed "header" to "block".
+        NOTE: moodified a lot
+        """
+
+        try:
+            first_block = first(blocks)
+        except StopIteration:
+            return tuple(), tuple()
+        else:
+            for parent, child in sliding_window(2, blocks):
+                if parent.hash != child.parent_hash:
+                    raise ValidationError(
+                        "Non-contiguous chain. Expected {} to have {} as parent but was {}".format(
+                            encode_hex(child.hash),
+                            encode_hex(parent.hash),
+                            encode_hex(child.parent_hash),
+                        )
+                    )
+
+            is_genesis = first_block.parent_hash == GENESIS_PARENT_HASH
+            if not is_genesis and not cls._block_exists(db, first_block.parent_hash):
+                raise ParentNotFound(
+                    "Cannot persist block ({}) with unknown parent ({})".format(
+                        encode_hex(first_block.hash), encode_hex(first_block.parent_hash)))
+
+            score = 0 if is_genesis else cls._get_score(db, first_block.parent_hash)
```

Comment:
Preference would be to expand this to multiline `if/else` as I think it is easier to read.